File: CeaserCipher.py


def enycrypt(str,key):
    result = ""
    for i in range(len(str)):
        if str[i].isupper():
            result+=chr((ord(str[i])+key-65)%26+65)
        elif str[i].islower():
           result+=chr((ord(str[i])+key-97)%26+97)
        else:
            result+=str[i]
    return result

def decrypt(str,key):
    result = ""
    for i in range(len(str)):
        if str[i].isupper():
            result+=chr((ord(str[i])-key-65)%26+65)
        elif str[i].islower():
           result+=chr((ord(str[i])-key-97)%26+97)
        else:
            result+=str[i]
    return result

flag=False
while True:
    option=int(input("Enter 1 to encrypt, 2 to decrypt, 3 to exit: "))
    if option==1:
        if flag is True:
            print("Already encrypted, please decrypt first")
        else:
            flag=True
            str=input("Enter the string: ")
            enycrypt_str=enycrypt(str,3)
            print("Encrypted string: ",enycrypt_str)
    elif option==2:
        if flag is not True:
            print("Please encrypt first")
        else:
            flag=False
            decrypt_str=decrypt(enycrypt_str,3)
            print("Decrypted string: ",decrypt_str)
    else:
        break

File: CeaserCipher.cpp

#include <bits/stdc++.h>
using namespace std;
void decrypt(string &text, int key)
{
    cout << "Encrypted String:-  " << text << endl;
    for (int i = 0; i < text.length(); i++)
    {
        if (text[i] >= 'a' && text[i] <= 'z')
        {
            int temp = text[i] - 'a';
            temp = (temp - key) % 26;
            if (temp < 0)
            {
                temp = temp + 26;
            }
            text[i] = temp + 'a';
        }
        else if (text[i] >= 'A' && text[i] <= 'Z')
        {
            int temp = text[i] - 'A';
            temp = (temp - key) % 26;
            if (temp < 0)
            {
                temp = temp + 26;
            }
            text[i] = temp + 'A';
        }
    }
    cout << "Decrypted String:-  " << text << endl;
}
void enycrypt(string &text, int key)
{
    cout << "Given String:-  " << text << endl;
    for (int i = 0; i < text.length(); i++)
    {
        if (text[i] >= 'a' && text[i] <= 'z')
        {
            int temp = text[i] - 'a';
            temp = (temp + key) % 26;
            text[i] = temp + 'a';
        }
        else if (text[i] >= 'A' && text[i] <= 'Z')
        {
            int temp = text[i] - 'A';
            temp = (temp + key) % 26;
            text[i] = temp + 'A';
        }
    }
    cout << "Encrypted String:-  " << text << endl;
}
int main()
{
    string text;
    cout << "Enter the string:-  ";
    getline(cin, text);
    int key;
    cout << "Enter the key:-  ";
    cin >> key;
    bool flag = false;
    while (1)
    {
        int option;
        cout << "1. Encrypt\n2. Decrypt\n3. Exit\n";
        cin >> option;
        if (option == 1)
        {
            flag = true;
            enycrypt(text, key);
        }
        else if (option == 2)
        {
            if (!flag)
            {
                cout << "Please Encrypt the string first\n";
                continue;
            }
            decrypt(text, key);
            flag = false;
        }
        else if (option == 3)
        {
            break;
        }
        else
        {
            cout << "Invalid Option\n";
        }
    }
    return 0;
}